bundle/config/mutator/python/apply_python_mutator.go

```diff
@@ -0,0 +1,275 @@
+	"github.com/databricks/cli/libs/dyn/yamlloader"
+	"github.com/databricks/cli/libs/log"
+	"github.com/databricks/cli/libs/process"
+)
+
+type phase string
+
+const (
+	// ApplyPythonMutatorPhasePreInit is the phase that while bundle configuration is loading.
+	//
+	// At this stage, PyDABs adds statically defined resources to the bundle configuration.
+	// Which resources are added should be deterministic and not depend on the bundle configuration.
+	//
+	// We also open for possibility of appending other sections of bundle configuration,
+	// for example, adding new variables. However, this is not supported yet, and CLI rejects
+	// such changes.
+	ApplyPythonMutatorPhasePreInit phase = "preinit"
+
+	// ApplyPythonMutatorPhaseInit is the phase after bundle configuration was loaded, and
+	// the list of statically declared resources is known.
+	//
+	// At this stage, PyDABs adds resources defined using generators, or mutates existing resources,
+	// including the ones defined using YAML.
+	//
+	// During this process, within generator and mutators, PyDABs can access:
+	// - selected deployment target
+	// - bundle variables values
+	//
+	// The following is not available:
+	// - variables referencing other variables are in unresolved format
+	// - variables provided through CLI arguments
+	//
+	// PyDABs can output YAML containing references to variables, and CLI should resolve them.
+	//
+	// Existing resources can't be removed, and CLI rejects such changes.
+	ApplyPythonMutatorPhaseInit phase = "init"
+)
+
+type applyPythonMutator struct {
+	phase phase
+}
+
+func ApplyPythonMutator(phase phase) bundle.Mutator {
+	return &applyPythonMutator{
+		phase: phase,
+	}
+}
+
+func (m *applyPythonMutator) Name() string {
+	return fmt.Sprintf("ApplyPythonMutator(%s)", m.phase)
+}
+
+func getExperimental(b *bundle.Bundle) config.Experimental {
+	if b.Config.Experimental == nil {
+		return config.Experimental{}
+	}
+
+	return *b.Config.Experimental
+}
+
+func (m *applyPythonMutator) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	experimental := getExperimental(b)
+
+	if !experimental.PyDABs.Enable {
+		return nil
+	}
+
+	if experimental.PyDABs.Enable && experimental.VEnv.Path == "" {
+		return diag.Errorf("\"experimental.pydabs.enable\" can only be used when \"experimental.venv.path\" is set")
+	}
+
+	err := b.Config.Mutate(func(leftRoot dyn.Value) (dyn.Value, error) {
+		pythonPath := interpreterPath(experimental.VEnv.Path)
+
+		if _, err := os.Stat(pythonPath); err != nil {
+			if os.IsNotExist(err) {
+				return dyn.InvalidValue, fmt.Errorf("can't find %q, check if venv is created", pythonPath)
+			} else {
+				return dyn.InvalidValue, fmt.Errorf("can't find %q: %w", pythonPath, err)
+			}
+		}
+
+		rightRoot, err := m.runPythonMutator(ctx, b.RootPath, pythonPath, leftRoot)
+
+		if err != nil {
+			return dyn.InvalidValue, err
+		}
+
+		visitor, err := createOverrideVisitor(ctx, m.phase)
+
+		if err != nil {
```

Comment:
Nit: could you please remove the newlines between assignment and err-checks? I find this harder to read as they all look disjoint while they actually belong together.